File: src/models/week-menu.ts
import type { FoodAllergen } from "./food-allergen";
import type { FoodLabel } from "./food-label";
import type { Menu } from "./menu";

export type WeekMenu = Readonly<{
  containsLunch: boolean
  containsDinner: boolean

  /**
   * Menu for each day of the week.
   */
  days: ReadonlyArray<Menu>

  /**
   * Week numbers that are available
   * for the menu.
   */
  weeks: ReadonlyArray<number>

  allergens: ReadonlyArray<FoodAllergen>
  labels: ReadonlyArray<FoodLabel>
}>;

File: library/src/commonMain/kotlin/models/WeekMenu.kt
package models

data class WeekMenu(
    val containsLunch: Boolean,
    val containsDinner: Boolean,

    /**
     * Menu for each day of the week.
     */
    val days: List<Menu>,

    /**
     * Week numbers that are available
     * for the menu.
     */
    val weeks: List<Int>,

    val allergens: List<FoodAllergen>,
    val labels: List<FoodLabel>
)